### PR TITLE
Fix integration tests.

### PR DESCRIPTION
**Pull request by zmoore-usgs**
Note that Travis will no longer work due to Artifactory being down and so I have disabled it for now by renaming the file.

Changes include:

1. Fixes for tests to account for new unique station IX constraints. The tests here were not updated after we began forcing station IX uniqueness in the ~~DB~~ legacy CRU service, and as a result some were failing because they were sharing the same station name.

2. Fixes configuration to account for new role mapping behavior (removing ROLE_ prefix)

3. Fixes configuration for service upgraded to spring boot 2.x to use the new jwkKeySet URI from WaterAuth instead of the old token_key

4. Fixes generation of random site numbers that were not always properly formatted.

5. Fixes issues with certificates generated for the services where the subjectAltName was not included and the hosts were not valid by updated python requests library standards.

6. Fixes an issue where using lots of volume mounts for certificates can result in the system running out of available mounting points for a given file causing the file to not be mounted in the container (fixed by switching to docker-compose secrets instead of volume mounts for most certificates)

### CODE REVIEW

**Comment by kmschoep-usgs** [comment]
We don't actually force uniqueness of STATION_IX in the database, it's handled by MLR-Legacy-Service.

**Comment by zmoore-usgs** [comment]
Ah yeah, that's right. Still coincides with the behavior I was seeing, but yeah forgot that we don't enforce that at the DB level.

**Comment by cschroed-usgs** [comment]
Should .travis.yml.old be deleted?

File: create_certificates.sh [hunk 1]
```diff
@@ -1,37 +1,21 @@
...
+openssl req -nodes -newkey rsa:2048 -keyout $CERT_DIR/ssl.key -out $CERT_DIR/ssl.csr \
+    -subj "/C=US/ST=Wisconsin/L=Middleon/O=US Geological Survey/OU=WMA/CN=mlr-int"
+openssl x509 -req \
+  -days 9999 -in $CERT_DIR/ssl.csr -signkey $CERT_DIR/ssl.key  -out $CERT_DIR/ssl.crt
+
+openssl genrsa -out $CERT_DIR/water-auth.key 2048
+openssl req -nodes -newkey rsa:2048 -keyout $CERT_DIR/water-auth.key -out $CERT_DIR/water-auth.csr \
+    -subj "/C=US/ST=Wisconsin/L=Middleon/O=US Geological Survey/OU=WMA/CN=water-auth"
+openssl x509 -req \
+  -days 9999 -in $CERT_DIR/water-auth.csr -signkey $CERT_DIR/water-auth.key  -out $CERT_DIR/water-auth.crt
```
Comment (cschroed-usgs) [inline create_certificates.sh:21]: Nice simplification!

**Comment by zmoore-usgs** [comment]
Re: Travis - Maybe not quite yet? I'm hopeful that the CHS Artifactory public URL Docker Image pulling woes will be resolved eventually and we can re-enable that file by just renaming it back to .travis.yml and then tweaking the URLs.

File: docker-compose-services.yml [hunk 2]
```diff
@@ -1,10 +1,37 @@
...
+
+  x-commons-secrets-python:
+    &python-secrets
+    secrets:
+      - source: ssl_crt
+        target: /home/python/certificates/ssl.crt
+      - source: ssl_key
+        target: /home/python/certificates/ssl.key
+      - source: water_auth_crt
+        target: /home/python/certificates/water-auth.crt
```
Comment (cschroed-usgs) [inline docker-compose-services.yml:24]: Good use of YAML merge keys! This config file is pretty DRY!

**Comment by cschroed-usgs** [comment]
> Re: Travis - Maybe not quite yet? I'm hopeful that the CHS Artifactory public URL Docker Image pulling woes will be resolved eventually and we can re-enable that file by just renaming it back to .travis.yml and then tweaking the URLs.

Yeah, I get that. I tend to prefer using old commits rather than .backup/.old files, but since this problem might get resolved soon, it might be fine to have this extra file in there for a few commits. Not a blocker.

I appreciated you writing a list summarizing the changes in the pull request. That kind of summary helps orient my review in pull requests when there are non-trivial changes across more than 10-ish files. You may have already thought of this, but another way to structure contributions is to put all of the changes related to each bullet point in a commit dedicated to just that kind of change, and then encourage reviewers to go through commit-by-commit. It doesn't make sense to use that approach in all cases, or necessarily even in this one, but I wanted to make sure you knew about this tool in your toolbox going forward.

**Pull request merged**